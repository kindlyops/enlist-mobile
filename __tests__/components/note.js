import React from 'react'
import renderer from 'react-test-renderer'
import { fromJS } from 'immutable'
import moment from 'moment'
import Note from '../../app/components/application/note'

describe('Initial state', () => {
  test('basics', () => {
    let note = fromJS({
      id: 1,
      linkedText: 'Here is a note',
      createdAt: moment('2019-01-01T12:00:00+00:00')
    })

    let user = fromJS({
      id: 1,
      fullName: 'Test User'
    })

    const tree = renderer.create(<Note note={note} createdBy={user} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
