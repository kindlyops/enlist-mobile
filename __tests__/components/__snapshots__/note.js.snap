// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial state basics 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#EEEEEE",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginTop": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "padding": 10,
      }
    }
  >
    Here is a note
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FCFCFC",
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginTop": 5,
        "padding": 10,
        "paddingBottom": 7,
        "paddingTop": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#90959D",
            "fontSize": 12,
          },
        ]
      }
    >
      Test User
       on 
      Jan 1, 2019
    </Text>
  </View>
</View>
`;
